src/start/RealmScreen.js

```diff
@@ -0,0 +1,83 @@
+import React from 'react';
+import { TextInput } from 'react-native';
+import { bindActionCreators } from 'redux';
+import { connect } from 'react-redux';
+
+import { styles, Screen, ErrorMsg, Button } from '../common';
+import { getAuthBackends } from '../api/apiClient';
+import config from '../config';
+import { realmAdd } from '../account/accountActions';
+import AuthBackendPicker from './AuthBackendPicker';
+
+type Props = {
+  realm: ?string,
+  navigateTo: (route: string) => void,
+}
+
+class RealmScreen extends React.Component {
+
+  props: Props;
+
+  constructor(props: Props) {
+    super(props);
+
+    const realmFromConfig = process.env.NODE_ENV === 'development' ? config.devRealm : config.productionRealm;
+    this.state = {
+      progress: false,
+      realm: props.realm || realmFromConfig,
+      authBackends: [],
+    };
+  }
+
+  tryRealm = async () => {
+    const { realm } = this.state;
+
+    this.setState({ progress: true, error: '' });
+
+    try {
+      const authBackends = await getAuthBackends({ realm });
```

Comment:
Thanks for the link. I'm not sure I agree that the state should live locally in this component in this case, but we can chat about that later